sqlmesh/core/test/definition.py

```diff
@@ -111,13 +110,8 @@ def assert_equal(self, expected: pd.DataFrame, actual: pd.DataFrame) -> None:
                 check_datetimelike_compat=True,
             )
         except AssertionError as e:
-            diff = "\n".join(
-                difflib.ndiff(
-                    [str(x) for x in expected.to_dict("records")],
-                    [str(x) for x in actual.to_dict("records")],
-                )
-            )
-            e.args = (f"Data differs\n{diff}",)
+            diff = expected.compare(actual).rename(columns={"self": "expected", "other": "actual"})
```

Comment:
Would it make sense to transpose beyond a certain number of columns?

So diff.T

Reply:
Maybe we should just transpose by default. Are you concerned about producing cluttered output for wide tables due to wrapping up the lines?

Reply:
For example:
```
F.
======================================================================
FAIL: test_order_items (sushi/tests/test_order_items.yaml)
----------------------------------------------------------------------
AssertionError: Data differs

                            0
id       expected        10.0
         actual           1.0
order_id expected        10.0
         actual           1.0
item_id  expected        10.0
         actual           2.0
quantity expected        10.0
         actual           4.0
ds       expected  2022-01-06
         actual    2022-01-01

----------------------------------------------------------------------
Ran 2 tests in 0.063s

FAILED (failures=1)
```

Reply:
OTOH if there are many rows that differ than we'll get the same line wrapping

Reply:
FWIW I'm not sure if the previous approach (i.e. using `difflib`) addressed these issues gracefully either. It would just spit out the whole rows which could line wrap pretty badly and also include unnecessary columns for which the actual and expected values matched.

Reply:
Nvm I forgot that Pandas doesn't wrap the lines but instead just hides the columns:

```
➜  test sqlmesh test
F
======================================================================
FAIL: test_example_full_model (tests/test_full_model.yaml)
----------------------------------------------------------------------
AssertionError: Data differs

   item_id        num_orders               a               b  ...      i        j               k               l
  expected actual   expected actual expected actual expected  ... actual expected actual expected actual expected actual
0      NaN    NaN        3.0    2.0     None      3     None  ...     11     None     12     None     13     None     14
1      1.0    2.0        NaN    NaN     None      3     None  ...     11     None     12     None     13     None     14

[2 rows x 28 columns]

----------------------------------------------------------------------
Ran 1 test in 0.024s

FAILED (failures=1)
```

@tobymao any thoughts on this?